File: src/main/java/com/example/probation/validatiion/annotation/UniqueUsername.java
package com.example.probation.validatiion.annotation;


import com.example.probation.validatiion.validator.NameValidator;

import javax.validation.Constraint;
import javax.validation.Payload;
import java.lang.annotation.*;

@Target({ElementType.METHOD, ElementType.FIELD})
@Retention(RetentionPolicy.RUNTIME)
@Documented
@Constraint(validatedBy = NameValidator.class)
public @interface UniqueUsername {
    String message() default "{username.not.unique}";

    Class<?>[] groups() default {};

    Class<? extends Payload>[] payload() default {};
}

File: src/main/kotlin/com/example/probation/validation/annotation/UniqueUsername.kt
package com.example.probation.validation.annotation

import com.example.probation.validation.validator.NameValidator
import com.example.probation.validation.validator.NewNameValidator
import javax.validation.Constraint
import javax.validation.Payload
import kotlin.reflect.KClass

@MustBeDocumented
@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.CLASS, AnnotationTarget.FIELD)
@Constraint(validatedBy = [NameValidator::class, NewNameValidator::class])
annotation class UniqueUsername(
    val message: String = "{username.not.unique}",
    val groups: Array<KClass<*>> = [],
    val payload: Array<KClass<out Payload>> = []
)
